recipes/publish.rb

```diff
@@ -17,63 +17,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+project_secrets = get_project_secrets
+origin = 'delivery'
 
-hab_pkgident = node['habitat-build']['hab-pkgident']
-hab_studio_pkgident = node['habitat-build']['hab-studio-pkgident']
-
-# e.g., `sample-verify-syntax`
-studio_slug = [
-  node['delivery']['change']['project'],
-  node['delivery']['change']['stage'],
-  node['delivery']['change']['phase']
-].join('-')
-
-# e.g., `/hab/studios/sample-verify-syntax`
-studio_path = ::File.join('/hab/studios', studio_slug)
+if habitat_origin_key?
+  keyname = project_secrets['habitat']['keyname']
+  origin = keyname.split('-')[0...-1].join('-')
+end
 
 # set local variables we're going to use in `lazy` properties later in
 # the chef run
 artifact = nil
-artifact_hash = nil
-artifact_pkgident = nil
+build_version = nil
 last_build_env = nil
+project_name = node['delivery']['change']['project']
 
-ruby_block 'build-plan' do
-  block do
-    Dir.chdir(node['delivery']['workspace']['repo'])
-    ENV['TERM'] = 'ansi'
-    command = "sudo #{::File.join('/hab/pkgs', hab_studio_pkgident, 'bin/hab-studio')}"
-    command << " -r /hab/studios/#{studio_slug}"
-    command << " build #{habitat_plan_dir}"
-
-    build = shell_out(command)
-
-    if build.exitstatus > 0
-      puts build.stdout
-      puts build.stderr
-      raise 'The plan.sh did NOT come together, bailing out!'
-    end
+execute 'build-plan' do
+  command "sudo #{hab_studio_binary}" \
+          " -r #{hab_studio_path}" \
+          " -k #{origin}" \
+          " build #{habitat_plan_dir}"
+  env(
+    'TERM' => 'ansi'
+  )
+  cwd node['delivery']['workspace']['repo']
+  live_stream true
+end
```

Comment:
Is `TERM=ansi` the key here to resolving the bug when displaying the `live_stream` in Delivery?